modulus/utils/filesystem.py

```diff
@@ -71,6 +151,9 @@ def _download_cached(path: str, recursive: bool = False) -> str:
         if path.startswith("s3://"):
             fs = _get_fs(path)
             fs.get(path, cache_path, recursive=recursive)
+        elif path.startswith("ngc://"):
+            path = _download_ngc_model_file(path, cache_path)
```

Comment:
Maybe add timeout input here and have timeout be given to `_download_cached`. Not a big deal though.

Reply:
Considered this, but then you would then have to realistically also control the fsspec timeout too and even potentially pass it up to the Package class.

Anyway, I just left it as a controllable input now so we can do this in the future if really needed. :+1: